securedrop-admin

```diff
@@ -50,42 +50,119 @@ def sdconfig(args):
                             cwd=ANSIBLE_PATH, env=os.environ.copy())
 
 
+def run_command(command):
+    """
+    Wrapper function to display stdout for running command,
+    similar to how shelling out in a Bash script displays rolling output.
+
+    Yields a list of the stdout from the `command`, and raises a
+    CalledProcessError if `command` returns non-zero.
+    """
+    popen = subprocess.Popen(command, stdout=subprocess.PIPE)
+    for stdout_line in iter(popen.stdout.readline, ""):
+        yield stdout_line
+    popen.stdout.close()
+    return_code = popen.wait()
+    if return_code:
+        raise subprocess.CalledProcessError(return_code, command)
+
+
+def install_apt_dependencies(args):
+    """
+    Install apt dependencies in Tails. In order to install Ansible in
+    a virtualenv, first there are a number of Python prerequisites.
+    """
+    sdlog.info("Installing SecureDrop Admin dependencies")
+    sdlog.info(("You'll be prompted for the temporary Tails admin password,"
+                " which was set on Tails login screen"))
+
+    apt_command = ['sudo', 'su', '-c',
+                   "apt-get update && \
+                   apt-get install -y \
+                   python-virtualenv \
+                   python-pip \
+                   ccontrol \
+                   virtualenv \
+                   libffi-dev \
+                   libssl-dev \
+                   libpython2.7-dev",
+                   ]
+
+    try:
+        # Print command results in real-time, to keep Admin apprised
+        # of progress during long-running command.
+        for output_line in run_command(apt_command):
+            print(output_line.rstrip())
+    except subprocess.CalledProcessError:
+        # Tails supports apt persistence, which was used by SecureDrop
+        # under Tails 2.x. If updates are being applied, don't try to pile
+        # on with more apt requests.
+        sdlog.error(("Cannot lock apt. Tails may be performing automatic"
```

Comment:
Good feedback—I struggled with proper resolution here. Seems like simple saying that apt failed is indeed the most informative approach, so I'll switch to that.